### PR TITLE
Remove unused declarations

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

File: lib/libutee/include/user_ta_header.h [hunk 1]
```diff
@@ -115,9 +115,6 @@ extern struct __elf_phdr_info __elf_phdr_info;
 #define TA_PROP_STR_STACK_SIZE		"gpd.ta.stackSize"
 #define TA_PROP_STR_VERSION		"gpd.ta.version"
 #define TA_PROP_STR_DESCRIPTION		"gpd.ta.description"
-#define TA_PROP_STR_UNSAFE_PARAM	"op-tee.unsafe_param"
```
Comment (jenswi-linaro) [inline lib/libutee/include/user_ta_header.h:118]: I think the idea here was to report to the TA some of its flags as properties. It seems that it wasn't completely implemented though.

Comment (jenswi-linaro) [reply lib/libutee/include/user_ta_header.h:118]: @etienne-lms, what do you think? Do we care about these?

Comment (etienne-lms) [reply lib/libutee/include/user_ta_header.h:118]: I would say no, we don't care.
I even didn't remember we had these macros.
Maybe we should get rid of all of these TA_PROP_STR_*?

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: lib/libutee/include/user_ta_header.h (hunk 1 above)
Comment (jenswi-linaro) [reply lib/libutee/include/user_ta_header.h:118]: > I would say no, we don't care.

OK, agree.

> I even didn't remember we had these macros.

Me too! :-)

> Maybe we should get rid of all of these TA_PROP_STR_*?

No, the other are used and required by GP.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: lib/libutee/include/user_ta_header.h (hunk 1 above)
Comment (etienne-lms) [reply lib/libutee/include/user_ta_header.h:118]: > > Maybe we should get rid of all of these TA_PROP_STR_*?
>
> No, the other are used and required by GP.

Yes, sorry, I missed occurrences in **ta/arch/arm/user_ta_header.c**.

**Pull request merged**